mono/mini/mini-x86.h

```diff
@@ -350,5 +350,10 @@ mono_x86_start_gsharedvt_call (GSharedVtCallInfo *info, gpointer *caller, gpoint
 CallInfo*
 mono_arch_get_call_info (MonoMemPool *mp, MonoMethodSignature *sig);
 
+#define MONO_ARCH_AOT_ICALLS                           \
```

Comment:
The previous approach was fine, why do these need to be inlined ?

Reply:
Just keeping it optimal where easy.
You prefer the per-arch helper function, nested switch instead of one switch?
I can put that back and fill in the other architectures.

Reply:
Registration is also not terrible, then in the switch just index into the array.

Reply:
I prefer the helper function or the registration, this is only run a few times on startup, there is no need to make it optimal.

Reply:
I'm fine either way. The helper function could also be inline in the .h.
Anything is better than a hashtable imho.
Optimal is also smaller and pays recurring dividends, albeit small ones.

I think when I first did this I didn't realize the connections, so I had redundantly optimized registration, and then didn't use the registration.

The registration also doesn't need the strings.

Note that I removed the ifdef gsharedvt from the helpers, that was in registration.
It isn't really conditional, as the current system is configured.